apps/api-wallet/src/app/transactions/transactions.service.ts

```diff
@@ -49,12 +50,33 @@ export class TransactionsService {
     return history;
   }
 
-  async getHistoryByUser(userID: number): Promise<any | undefined> {
+  async getHistoryByUser(
+    userID: number,
+    page: number,
+    limit: number
+  ): Promise<any | undefined> {
+    const startIndex = (page - 1) * limit;
+    const endIndex = page * limit;
+
     const u_history = await this.databaseService.getQueryResult(
       'history_transaction',
       [userID]
     );
-    return u_history;
+
+    if (!u_history.length) throw new NotFoundException();
+
+    const paginatedTransactions = u_history.slice(startIndex, endIndex);
+
+    const hasNextPage = endIndex < u_history.length;
+
+    return {
+      totalTransactions: u_history.length,
+      currentPage: +page++,
```

Comment:
why did you do this for the currentPage?

Reply:
sa link for nextpage man gud kay maconvert na si page into string, which is maka cause og undesired output..then mao na ako na lang gibutangan og + operator sa start para magstay sya as number , then yung ++ para mo.increment siya when clicking to next page

Reply:
hmm let's not do this, for readability add large +page sa currentPage and put the increment sa nextPage na

Reply:
dli ko add og increment sa nextpage bai since boolean iya type..Naa ra koy fixed, pakicheck na lang..
Med-u samok lang bai ang bag.o push kay naapil ag testing file..